content/5_Plat/Sweep_Line.mdx

```diff
@@ -173,9 +173,82 @@ For each point, we recalculate $\delta = \min(\delta, \delta_p)$, and update our
 
 <FocusProblem problem="manSam" />
 
-(KACTL code)
+The key observation is that although there are many points, they are spread across a small surface i.e. $0 \leq x, y \le 1000$. So instead of using Kruskal's or Prim's
+algorithm, we can use Dijkstra's. Starting from an arbitrary point, from the first one in the code below, run Dijkstra's algorithm. The priority queue
+will sort the points after the distance to the MST.
 
-explanation? topcoder prob has
+*Time Complexity:* $\mathcal{O}(1)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int NMAX = 1000;
+const int dx[4] = {-1, 0, 1, 0};
+const int dy[4] = {0, 1, 0, -1};
+
+int d[NMAX][NMAX];
+bool mat[NMAX][NMAX];  // mat[x][y] = if (x, y) point is in the input
+
+int main() {
+	int n;
+	cin >> n;
+	memset(d, -1, sizeof(d));
+	priority_queue<pair<int, pair<int, int>>, vector<pair<int, pair<int, int>>>,
+	               greater<pair<int, pair<int, int>>>>
+	    pq;
```

Comment:
formatting kinda off ?

Reply:
i think it's just the autoformatter being dumb

Reply:
yep it's the autoformatter

Reply:
Chaged